libraries/AP_HAL_ChibiOS/hwdef/common/stm32_util.c

```diff
@@ -229,29 +231,35 @@ void set_rtc_backup(uint8_t n, uint32_t v)
 #else
     PWR->CR1 |= PWR_CR1_DBP;
 #endif
-    ((__IO uint32_t *)&RTC->BKP0R)[n] = v;
+    while (n--) {
+        ((__IO uint32_t *)&RTC->BKP0R)[idx++] = *v++;
+    }
 }
 
 // see if RTC registers is setup for a fast reboot
 enum rtc_boot_magic check_fast_reboot(void)
 {
-    return (enum rtc_boot_magic)get_rtc_backup(0);
+    uint32_t v;
+    get_rtc_backup(0, &v, 1);
+    return (enum rtc_boot_magic)v;
 }
 
 // set RTC register for a fast reboot
 void set_fast_reboot(enum rtc_boot_magic v)
 {
-    set_rtc_backup(0, v);
+    uint32_t vv = (uint32_t)v;
+    set_rtc_backup(0, &vv, 1);
 }
 
 #else // NO_FASTBOOT
 
-// set RTC backup register 1
```

Comment:
`set_fast_reboot` should be being protected here.  Without `set_rtc_backup` hold-in-bootloader won't work.  Does this compile with `NO_FASTBOOT` set?